lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -124,11 +112,13 @@ public function getExtension()
      */
     protected function getFilename($id)
     {
+        $hash = hash('sha256', $id);
```

Comment:
What? Collisions on sha256? Please read http://stackoverflow.com/questions/4014090/is-it-safe-to-ignore-the-possibility-of-sha-collisions-in-practice for example.

> This approach can only work if the subclasses store the cache ID inside the stored file, and then check against it on retrieval. Probably the same is to be done also for deletions.

No idea what you are talking about.